datasets/swedish_medical_ner/swedish_medical_ner.py

```diff
@@ -0,0 +1,198 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""SwedMedNER: A Named Entity Recognition Dataset on medical texts in Swedish"""
+
+
+import re
+
+import datasets
+
+
+_CITATION = """\
+@inproceedings{almgrenpavlovmogren2016bioner,
+  title={Named Entity Recognition in Swedish Medical Journals with Deep Bidirectional Character-Based LSTMs},
+  author={Simon Almgren, Sean Pavlov, Olof Mogren},
+  booktitle={Proceedings of the Fifth Workshop on Building and Evaluating Resources for Biomedical Text Mining (BioTxtM 2016)},
+  pages={1},
+  year={2016}
+}
+"""
+
+
+_DESCRIPTION = """\
+SwedMedNER is a dataset for training and evaluating Named Entity Recognition systems on medical texts in Swedish.
+It is derived from medical articles on the Swedish Wikipedia, Läkartidningen, and 1177 Vårdguiden.
+"""
+
+
+_LICENSE = """\
+Creative Commons Attribution-ShareAlike 4.0 International Public License (CC BY-SA 4.0)
+See http://creativecommons.org/licenses/by-sa/4.0/ for the summary of the license.
+"""
+
+
+_URL = "https://github.com/olofmogren/biomedical-ner-data-swedish"
+
+
+_DATA_URL = "https://raw.githubusercontent.com/olofmogren/biomedical-ner-data-swedish/master/"
+
+
+class SwedishMedicalNerConfig(datasets.BuilderConfig):
+    """BuilderConfig for SwedMedNER"""
+
+    def __init__(self, **kwargs):
+        """
+        Args:
+          **kwargs: keyword arguments forwarded to super.
+        """
+        super(SwedishMedicalNerConfig, self).__init__(**kwargs)
+
+
+class SwedishMedicalNer(datasets.GeneratorBasedBuilder):
+    """SwedMedNER: A Named Entity Recognition Dataset on medical texts in Swedish"""
+
+    VERSION = datasets.Version("1.0.0")
+
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(name="wiki", version=VERSION, description="The Swedish Wikipedia part of the dataset"),
+        datasets.BuilderConfig(name="lt", version=VERSION, description="The Läkartidningen part of the dataset"),
+        datasets.BuilderConfig(name="1177", version=VERSION, description="The 1177 Vårdguiden part of the dataset"),
+    ]
+
+    def _info(self):
+        if self.config.name == "wiki":
+            features = datasets.Features(
+                {
+                    "sid": datasets.Value("string"),
+                    "sentence": datasets.Value("string"),
+                    "entities": datasets.Sequence(
+                        {
+                            "start": datasets.Value("int32"),
+                            "end": datasets.Value("int32"),
+                            "text": datasets.Value("string"),
+                            "type": datasets.Value("string"),
```

Comment:
You can use the ClassLabel feature type here:

```suggestion
                            "type": datasets.ClassLabel(names=["Disorder and Finding", "Pharmaceutical Drug", "Body Structure"]),
```

after this change you'll have to regenerate the dataset_infos.json file

Reply:
Done.